src/mainwindow.cpp

```diff
@@ -70,12 +76,39 @@ void MainWindow::detectTranslations()
                 }
             }
             m_translations.insert(langCode, {qtTranslator, appTranslator});
-            QSettings settings(QStringLiteral("%1/settings.ini").arg(QCoreApplication::applicationDirPath()), QSettings::IniFormat);
-            QString savedLang = settings.value("language", QStringLiteral("en")).toString();
-            if (savedLang == langCode) {
+
+            if (m_settings->value("language", QStringLiteral("en")).toString() == langCode) {
                 QApplication::installTranslator(qtTranslator);
                 QApplication::installTranslator(appTranslator);
             }
         }
     }
+
+    for (const QString &translation : langList) {
+        QTranslator translator;
+        translator.load(translation, i18nDir.absolutePath());
+        QString lang = translation.mid(13,2);
+        auto langAction = ui->menuLanguage->addAction(translator.translate("MainWindow","TRANSLATE TO YOUR LANGUAGE NAME"));
+        langAction->setData(lang);
+        langAction->setCheckable(true);
+        langAction->setChecked(m_settings->value("language").toString()==lang);
+    }
+}
+
+void MainWindow::changeLanguage(QAction *action)
+{
+    m_settings->setValue("language",action->data());
+    const QList<QAction*> allActions = ui->menuFile->findChildren<QAction*>();
+    for (QAction *other : allActions) {
+        other->setChecked(false);
+    }
+    action->setChecked(true);
+    QTranslator translator;
+    QString title,message;
+    translator.load(QString("FrogArchiver_%1.qm").arg(m_settings->value("language").toString()), QString("%1/i18n").arg(QCoreApplication::applicationDirPath()));
+    title= translator.translate("MainWindow","Language Changed");
+    if(title.isEmpty()){title = "Language Changed";}
+    message = translator.translate("MainWindow","You must restart for the language to change.");
+    if(message.isEmpty()){message = "You must restart for the language to change.";}
+    QMessageBox::information(this,title,message);
```

Comment:
I am curious, is it possible just auto restart the app after the language changes?

Reply:
Or at least we give prompt dialog and the "OK" button will restart the app.

Reply:
Not sure about restart we can have the application close easy enough

Reply:
I found this: https://wiki.qt.io/How_to_make_an_Application_restartable

What do you think?

Reply:
Not a Fan .. Restarts are hacky..

Reply:
Okay. :smile: If you think this PR is ready. I'll merge this PR. Thanks.